File: app/src/main/java/com/example/task2/models/BenchmarkModule.java
package com.example.task2.models;

import com.example.task2.models.benchmarks.Benchmark;
import com.example.task2.models.benchmarks.CollectionBenchmark;
import com.example.task2.models.benchmarks.MapBenchmark;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class BenchmarkModule {

    @Provides
    @Named("CollectionBenchmark")
    public Benchmark provideCollectionBenchmark() {
        return new CollectionBenchmark();
    }

    @Provides
    @Named("MapBenchmark")
    public Benchmark provideMapBenchmark() {
        return new MapBenchmark();
    }
}

File: app/src/main/java/com/example/task2/models/BenchmarkModule.kt
package com.example.task2.models

import com.example.task2.models.benchmarks.Benchmark
import com.example.task2.models.benchmarks.CollectionBenchmark
import com.example.task2.models.benchmarks.MapBenchmark
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class BenchmarkModule {

    @Provides
    @Named("CollectionBenchmark")
    fun provideCollectionBenchmark(): Benchmark = CollectionBenchmark()

    @Provides
    @Named("MapBenchmark")
    fun provideMapBenchmark(): Benchmark = MapBenchmark()
}